### PR TITLE
`PyroTrainingPlan`

### PR DESCRIPTION
**Pull request by ordabayevy**

### CODE REVIEW

**Comment by ordabayevy** [comment]
@mbabadi can you review this pls?

**Comment by mbabadi** [comment]
@ordabayevy on it!

File: test/test_ppca.py [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
-from pyro import infer, optim
+from torch.utils.data import DataLoader, Dataset
 
 from scvid.module import ProbabilisticPCAPyroModule
+from scvid.train import PyroTrainingPlan
 
 n, g, k = 1000, 10, 2
 
 
+class dataset(Dataset):
```
Comment (mbabadi) [inline test/test_ppca.py:14]: `class dataset` => `class TestDataset`

File: test/test_ppca.py [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        return {"X": self.data[idx]}
+
+
 @pytest.fixture
 def x_ng():
     torch.manual_seed(1465)
```
Comment (mbabadi) [inline test/test_ppca.py:27]: Static RNG :D

File: scvid/train/training_plan.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+from typing import Optional
+
+import pyro
+import pytorch_lightning as pl
+import torch
+
+
+class PyroTrainingPlan(pl.LightningModule):
```
Comment (mbabadi) [inline scvid/train/training_plan.py:8]: Might want to add to the Docstring that this implementation is a stripped-down version of what is in scvi-tools and also provide a permalink reference.

**Review by mbabadi — APPROVED** [review]

File: test/test_ppca.py [hunk 4]
```diff
@@ -31,21 +44,22 @@ def test_probabilistic_pca(x_ng, minibatch, ppca_flavor, learn_mean):
...
-        if minibatch:
-            batch_size = n // 2
-            ind = torch.randint(n, (batch_size,))
-            x_batch = x_ng[ind]
-        else:
-            x_batch = x_ng
-        svi.step(x_batch)
+    training_plan = PyroTrainingPlan(ppca, optim_kwargs={"lr": 1e-2})
+    # trainer
+    trainer = pl.Trainer(accelerator="cpu", max_epochs=4000)
```
Comment (mbabadi) [inline test/test_ppca.py:60]: Out of curiosity, does PL _control_ the optimizer? e.g. if one wants to user a scheduler, would PL automatically give the optimizer to the scheduler?

**Review by mbabadi — CHANGES_REQUESTED** [review]

**Comment by mbabadi** [comment]
@ordabayevy I approved by mistake, hence, added a new review and requested changes. Sorry for the confusion!

File: test/test_ppca.py (hunk 4 above)
Comment (ordabayevy) [reply test/test_ppca.py:61]: Not sure, I will have to dig more into it to see how it is handled.

**Comment by ordabayevy** [comment]
@mbabadi the code is ready for another round of review :)

**Review by mbabadi — APPROVED** [review]
